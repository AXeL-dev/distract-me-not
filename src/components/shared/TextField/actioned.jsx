import React, { Component } from 'react';
import { Pane, TextInput, Button, Paragraph } from 'evergreen-ui';
import { RawHTML, OuterPane } from 'components';

class TextField extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: props.value || '',
    };
  }

  handleChange = (event) => {
    this.setState({ value: event.target.value });
    if (this.props.onChange) {
      this.props.onChange(event);
    }
  };

  handleKeyDown = (event) => {
    if (event.key === 'Enter') {
      this.submit();
    }
  };

  handleButtonClick = (event) => {
    this.submit();
  };

  submit = () => {
    if (this.props.onSubmit && !this.props.disabled) {
      const callback = (value) => this.setState({ value: value });
      this.props.onSubmit(this.state.value, callback);
    }
  };

  render() {
    return (
      <OuterPane {...this.props}>
        <Pane display="flex">
          <Pane display="flex" alignItems="center" flex={1}>
            <TextInput
              width="100%"
              value={this.state.value}
              onChange={this.handleChange}
              onKeyDown={this.handleKeyDown}
              placeholder={this.props.placeholder}
              required={this.props.required}
              borderTopRightRadius={this.props.hasButton ? 0 : 3}
              borderBottomRightRadius={this.props.hasButton ? 0 : 3}
            />
          </Pane>
          {this.props.hasButton && (
            <Pane display="flex" alignItems="center" marginLeft={-1}>
              <Button
                appearance={this.props.buttonAppearance || 'primary'}
                borderTopLeftRadius={0}
                borderBottomLeftRadius={0}
                onClick={this.handleButtonClick}
                disabled={this.props.disabled}
              >
                {this.props.buttonLabel || '+'}
              </Button>
            </Pane>
          )}
        </Pane>
        {this.props.hint && (
          <Paragraph size={300} color="muted" marginTop={6}>
            <RawHTML>{this.props.hint}</RawHTML>
          </Paragraph>
        )}
      </OuterPane>
    );
  }
}

export default TextField;
