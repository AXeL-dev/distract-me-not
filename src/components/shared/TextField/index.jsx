import LabeledTextField from './labeled';
import ActionedTextField from './actioned';

export function TextField(props) {
  return props.label ? <LabeledTextField {...props} /> : <ActionedTextField {...props} />;
}
