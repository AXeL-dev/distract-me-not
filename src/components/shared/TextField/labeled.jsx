import React from 'react';
import { Pane, TextInput, Position } from 'evergreen-ui';
import { TooltipLabel, OuterPane } from 'components';

function TextField(props) {
  return (
    <OuterPane display="flex" {...props}>
      <Pane display="flex" alignItems="center" flex={1}>
        <TooltipLabel
          text={props.label}
          size={props.labelSize}
          color={props.labelColor}
          className={props.labelClassName}
          tooltip={props.tooltip}
          tooltipPosition={props.tooltipPosition}
        />
      </Pane>
      <Pane
        display="flex"
        alignItems="center"
        width={props.width}
        marginRight={props.position === Position.LEFT ? 12 : 0}
      >
        <TextInput
          width={props.width}
          value={props.value}
          placeholder={props.placeholder}
          onChange={props.onChange}
          disabled={props.disabled}
        />
      </Pane>
    </OuterPane>
  );
}

export default TextField;
