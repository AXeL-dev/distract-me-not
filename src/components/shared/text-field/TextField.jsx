import { Component } from 'react';
import { Pane, TextInput, Button, Paragraph } from 'evergreen-ui';
import RawHTML from '../raw-html/RawHTML';

export default class TextField extends Component {

  constructor(props) {
    super(props);
    this.state = {
      value: props.value || ''
    };
  }

  handleChange = (event) => {
    this.setState({ value: event.target.value });
    if (this.props.onChange) {
      this.props.onChange(event);
    }
  }

  handleKeyDown = (event) => {
    if (event.key === 'Enter') {
      this.submit();
    }
  }

  handleButtonClick = (event) => {
    this.submit();
  }

  submit = () => {
    if (this.props.onSubmit) {
      const callback = (value) => this.setState({ value: value });
      this.props.onSubmit(this.state.value, callback);
    }
  }

  render() {
    return (
      <Pane
        padding={this.props.padding}
        paddingX={this.props.paddingX || this.props.padding}
        paddingY={this.props.paddingY || this.props.padding}
        paddingTop={this.props.paddingTop || this.props.paddingY}
        paddingBottom={this.props.paddingBottom || this.props.paddingY}
        paddingLeft={this.props.paddingLeft || this.props.paddingX}
        paddingRight={this.props.paddingRight || this.props.paddingX}
        margin={this.props.margin}
        marginX={this.props.marginX || this.props.margin}
        marginY={this.props.marginY || this.props.margin}
        marginTop={this.props.marginTop || this.props.marginY}
        marginBottom={this.props.marginBottom || this.props.marginY}
        marginLeft={this.props.marginLeft || this.props.marginX}
        marginRight={this.props.marginRight || this.props.marginX}
      >
        <Pane display="flex">
          <Pane display="flex" alignItems="center" flex={1}>
            <TextInput
              width="100%"
              value={this.state.value}
              onChange={this.handleChange}
              onKeyDown={this.handleKeyDown}
              placeholder={this.props.placeholder}
              disabled={this.props.disabled}
              required={this.props.required}
              borderTopRightRadius={this.props.hasButton ? 0 : 3}
              borderBottomRightRadius={this.props.hasButton ? 0 : 3}
            />
          </Pane>
          {this.props.hasButton &&
            <Pane display="flex" alignItems="center" marginLeft={-1}>
              <Button
                appearance={this.props.buttonAppearance || 'primary'}
                borderTopLeftRadius={0}
                borderBottomLeftRadius={0}
                onClick={this.handleButtonClick}
              >
                {this.props.buttonLabel ||Â '+'}
              </Button>
            </Pane>
          }
        </Pane>
        {this.props.hint &&
          <Paragraph size={300} color="muted" marginTop={6}>
            <RawHTML>{this.props.hint}</RawHTML>
          </Paragraph>
        }
      </Pane>
    );
  }
}
