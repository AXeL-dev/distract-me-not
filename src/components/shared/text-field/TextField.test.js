import { render, fireEvent } from "@testing-library/react";
import TextField from "./TextField";

it('renders correctly', () => {
  const { asFragment } = render(<TextField />);
  expect(asFragment()).toMatchSnapshot();
});

it('handles value change', () => {
  const handleChange = jest.fn();
  const { getByRole } = render(<TextField value="some text" onChange={handleChange} />);
  const input = getByRole('textbox');
  fireEvent.change(input, { target: { value: 'some other text' } })
  expect(handleChange).toHaveBeenCalledTimes(1);
});

it('handles submit using both enter key press & button click', () => {
  const handleSubmit = jest.fn();
  const { getByRole } = render(<TextField hasButton onSubmit={handleSubmit} />);
  const input = getByRole('textbox');
  fireEvent.keyDown(input, { key: 'Enter' });
  expect(handleSubmit).toHaveBeenCalledTimes(1);
  const button = getByRole('button');
  fireEvent.click(button);
  expect(handleSubmit).toHaveBeenCalledTimes(2);
});
