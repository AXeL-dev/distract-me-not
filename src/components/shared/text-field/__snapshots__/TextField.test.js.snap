// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-nil ub-box-szg_border-box"
  >
    <div
      class="css-nil ub-dspl_flex ub-box-szg_border-box"
    >
      <div
        class="css-nil ub-dspl_flex ub-algn-itms_center ub-flx_1 ub-box-szg_border-box"
      >
        <input
          aria-invalid="false"
          class="css-5ljhhe ub-color_425A70 ub-fnt-fam_b77syt ub-fnt-sze_12px ub-f-wght_400 ub-ln-ht_16px ub-ltr-spc_0 ub-w_100prcnt ub-h_32px ub-pl_10px ub-pr_10px ub-bblr_3px ub-bbrr_3px ub-btlr_3px ub-btrr_3px ub-box-szg_border-box"
          spellcheck="true"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
